internal/ethapi/api.go

```diff
@@ -752,8 +752,22 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumb
 // Call executes the given transaction on the state for the given block number.
 // It doesn't make and changes in the state/blockchain and is useful to execute and retrieve values.
 func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNr rpc.BlockNumber) (hexutil.Bytes, error) {
-	result, _, _, err := DoCall(ctx, s.b, args, blockNr, vm.Config{}, 5*time.Second)
-	return (hexutil.Bytes)(result), err
+	if gasCap := s.b.RPCGasCap(); gasCap != nil {
+		if new(big.Int).SetUint64(uint64(*args.Gas)).Cmp(gasCap) > 0 {
```

Comment:
args.Gas can be `nil`. This node will crash in that case. We handle gas illness inside `DoEstimateGas`. Perhaps we should pass the gascap as a parameter to `DoEstimateGas` and handle everything there.